core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/util/service/DeleteStatementGeneratorImpl.java

```diff
@@ -0,0 +1,389 @@
+/*
+ * #%L
+ * BroadleafCommerce Framework
+ * %%
+ * Copyright (C) 2009 - 2020 Broadleaf Commerce
+ * %%
+ * Licensed under the Broadleaf Fair Use License Agreement, Version 1.0
+ * (the "Fair Use License" located  at http://license.broadleafcommerce.org/fair_use_license-1.0.txt)
+ * unless the restrictions on use therein are violated and require payment to Broadleaf in which case
+ * the Broadleaf End User License Agreement (EULA), Version 1.1
+ * (the "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt)
+ * shall apply.
+ * 
+ * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the "Custom License")
+ * between you and Broadleaf Commerce. You may not use this file except in compliance with the applicable license.
+ * #L%
+ */
+package org.broadleafcommerce.core.util.service;
+
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.broadleafcommerce.common.util.dao.DynamicDaoHelperImpl;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+import javax.persistence.*;
+
+/**
+ * The idea of this class if to iterate over props of the passed class and if found prop with annotation @OneToMany
+ * recurse into it. In this way we build paths to the dependant classes. Once found "leaf" class that doesn't have
+ * any more dependant entities it will start to go back and build path for that classes. In a result we will have
+ * table name and path to that tabe from the root table, so we can build sql to delete dependencies in the right order.
+ * It knows how to deal with @Embedded, bidirectional @OneToMany, @OneToMany through @JoinTable
+ */
+@Service("blcDeleteStatementGenerator")
```

Comment:
I think this should be changed to "blDeleteStatementGenerator". There is extra 'c'